index.html

```diff
@@ -58,32 +50,29 @@
 
         const $ = {};
         window.$ = $;
+
+        // Create client configuration.
+        const configBuilder = Nimiq.Client.Configuration.builder();
         if (clientType === 'light') {
-            $.consensus = await Nimiq.Consensus.light();
-        } else if (clientType === 'nano') {
-            $.consensus = await Nimiq.Consensus.nano();
+            // Adding the Mempool feature triggers a light consensus.
+            configBuilder.feature(Nimiq.Client.Feature.MEMPOOL);
         }
 
-        $.blockchain = $.consensus.blockchain;
-        $.mempool = $.consensus.mempool;
-        $.network = $.consensus.network;
+        // Instantiate the client.
+        $.client = configBuilder.instantiateClient();
 
         // Generate a temporary wallet.
         $.wallet = await Nimiq.Wallet.generate();
+        document.getElementById('address').innerText = $.wallet.address.toUserFriendlyAddress();
 
-        // The nano client does not sync the full account info and cannot mine.
-        if (clientType !== 'nano') {
-            $.accounts = $.blockchain.accounts;
-            $.miner = new Nimiq.Miner($.blockchain, $.accounts, $.mempool, $.network.time, $.wallet.address);
-        }
+        $.client.addConsensusChangedListener(_onConsensusChanged);
 
-        $.consensus.on('established', () => _onConsensusEstablished());
-        $.consensus.on('lost', () => console.error('Consensus lost'));
+        $.client.addHeadChangedListener(_onHeadChanged);
 
-        $.blockchain.on('head-changed', () => _onHeadChanged());
-        $.network.on('peers-changed', () => _onPeersChanged());
+        // The Client API does not currently expose a peer-changed listener, so we need to use a workaround.
+        const consensus = await $.client._consensus;
+        consensus.network.on('peers-changed', _onPeersChanged);
```

Comment:
Hmpf. Can this be improved, i.e. add the `peers-changed` event, instead of documenting that we need a workaround? We're in control of the code, IMO it's a bit strange to say it requires a workaround when we could do it properly

Reply:
Alternatively, I'd suggest to drop the "peer count" feature from the demo

Reply:
I am changing it to update on an interval instead of using the private `._consensus` property, and also removed the mention of a workaround.